manifests/init.pp

```diff
@@ -437,6 +437,59 @@
   Hash               $mutators = {},
   ### END Hiera Lookups ###
 ) {
+
+  case $::osfamily {
+    'Darwin': {
+      $api_service_name  = 'org.sensuapp.sensu-api'
+      $etc_dir           = $sensu_etc_dir
+      $conf_dir          = "${etc_dir}/conf.d"
+      $user              = '_sensu'
+      $group             = 'wheel'
+      $home_dir          = '/opt/sensu'
+      $shell             = '/bin/false'
+      $dir_mode          = '0555'
+      $file_mode         = '0440'
+      $service_name      = 'org.sensuapp.sensu-client'
+    }
+    'Debian','RedHat': {
+      $api_service_name  = 'sensu-api'
+      $etc_dir           = $sensu_etc_dir
+      $conf_dir          = "${etc_dir}/conf.d"
+      $user              = $sensu_user  ? {
+        undef   => 'sensu',
+        default => $sensu_user,
+      }
+      $group        = $sensu_group ? {
+        undef   => 'sensu',
+        default => $sensu_group,
+      }
+      $home_dir          = '/opt/sensu'
+      $shell             = '/bin/false'
+      $dir_mode          = '0555'
+      $file_mode         = '0440'
+      $service_name      = 'sensu-client'
+    }
+    'windows': {
+      $api_service_name  = undef
+      $etc_dir           = $sensu_etc_dir
+      $conf_dir          = "${etc_dir}/conf.d"
+      $user              = $sensu_user  ? {
+        undef   => 'NT Authority\SYSTEM',
+        default => $sensu_user,
+      }
+      $group        = $sensu_group ? {
+        undef   => 'Administrators',
+        default => $sensu_group,
+      }
+      $home_dir          = $etc_dir
+      $shell             = undef
+      $dir_mode          = undef
+      $file_mode         = undef
+      $service_name      = 'sensu-client'
+    }
+    default: {}
```

Comment:
the default case should `fail("Detected kernel is <${::kernel}> and must be Darwin, Linux or Windows")`

I'm not 100% those are the actual kernel strings though.